tests/test_webhook.py

```diff
@@ -22,32 +24,98 @@ def test_webhook_doesnt_configure_sentry_if_dsn_not_present(
     expected_output = {
         "headers": {"Content-Type": "text/plain"},
         "isBase64Encoded": False,
         "statusCode": 405,
-        "body": "HTTP method not allowed. Only GET requests are supported.",
+        "body": "HTTP method DELETE not allowed. Supported methods: GET, POST.",
+    }
+    assert lambda_handler(request_data, {}) == expected_output
+    assert (
+        "lambdas.webhook",
+        30,
+        "Received invalid HTTP request method DELETE, returning 405 error response.",
+    ) in caplog.record_tuples
+
+
+def test_webhook_handles_get_request_missing_parameter(caplog):
+    request_data = {"requestContext": {"http": {"method": "GET"}}}
+    expected_output = {
+        "headers": {"Content-Type": "text/plain"},
+        "isBase64Encoded": False,
+        "statusCode": 400,
+        "body": "Malformed request, 'challenge' query parameter required.",
     }
     assert lambda_handler(request_data, {}) == expected_output
+    assert (
+        "lambdas.webhook",
+        30,
+        "Received GET request without 'challenge' query parameter, returning "
+        "400 error response.",
+    ) in caplog.record_tuples
 
 
-def test_webhook_handles_get_request_success(get_request):
+def test_webhook_handles_get_request_success(caplog, get_request):
     expected_output = {
         "headers": {"Content-Type": "text/plain"},
         "isBase64Encoded": False,
         "statusCode": 200,
         "body": "challenge-accepted",
     }
     assert lambda_handler(get_request, {}) == expected_output
+    assert "GET request received, returning 200 success response." in caplog.text
 
 
-def test_webhook_handles_get_request_missing_parameter(get_request_missing_parameter):
+def test_webhook_handles_post_request_invalid_signature(
+    caplog, post_request_invalid_signature
+):
     expected_output = {
         "headers": {"Content-Type": "text/plain"},
         "isBase64Encoded": False,
-        "statusCode": 400,
-        "body": "Malformed request, 'challenge' query parameter required",
+        "statusCode": 401,
+        "body": "Unable to validate signature. Has the webhook challenge secret "
+        "changed?",
+    }
+    assert lambda_handler(post_request_invalid_signature, {}) == expected_output
+    assert (
+        "lambdas.webhook",
+        30,
+        "Invalid signature in POST request, returning 401 error response.",
+    ) in caplog.record_tuples
+
+
+def test_webhook_handles_post_request_success(caplog, post_request_valid_signature):
+    expected_output = {
+        "headers": {"Content-Type": "text/plain"},
+        "isBase64Encoded": False,
+        "statusCode": 200,
+        "body": "Webhook POST request received and validated, no action taken.",
+    }
+    assert lambda_handler(post_request_valid_signature, {}) == expected_output
+    assert (
+        "POST request received and validated, no action triggered. Returning 200 "
+        "success response." in caplog.text
```

Comment:
Non-blocking and just curious, was the shift to `caplog.record_tuples` for some tests prompted by anything in particular?

Reply:
Yes! I specifically wanted to check the log level because I think we'll want to set up notifications for the warning logs, as those will indicate that something has gone wrong in a webhook post or that something other than our Alma instance attempted to post to the webhook. Either way we'll want to know.

Reply:
👍